server/handlers/handlers.go

```diff
@@ -30,25 +30,19 @@ func RegisterUser(w http.ResponseWriter, res *http.Request) {
 		return
 	}
 
-	//encrypt password
-	passDigest, hashErr := bcrypt.GenerateFromPassword([]byte(newUser.Password), 10)
-	if hashErr != nil {
-		respondWithError(w, "Something went wrong", http.StatusInternalServerError)
-		return
-	}
-	//temporarily print password, so that its not an unused variable
-	fmt.Printf("Encrypted Password: %v\n", passDigest)
+	err := data.CreateUser(newUser)
 
-	data.Users = append(data.Users, &newUser)
+	if err == nil {
```

Comment:
Instead of checking if there was no error and then doing the logic, should you check if there was an error, handle that case, and do the logic outside of an if-statement? Doing so would improve readability and reduce nesting.